nilearn/decomposition/multi_pca.py

```diff
@@ -266,31 +287,27 @@ def fit(self, imgs, y=None, confounds=None):
             self.masker_.fit()
         self.mask_img_ = self.masker_.mask_img_
 
-        parameters = get_params(MultiNiftiMasker, self)
-        # Remove non specific and redudent parameters
-        for param_name in ['memory', 'memory_level', 'confounds',
-                           'verbose', 'n_jobs']:
-            parameters.pop(param_name, None)
-
-        parameters['detrend'] = True
-
         # Now do the subject-level signal extraction (i.e. data-loading +
         # PCA)
-
+        if self.verbose:
+            print("[MultiPCA] Learning subject level PCAs")
         subject_pcas = Parallel(n_jobs=self.n_jobs, verbose=self.verbose)(
-            delayed(session_pca)(
+            delayed(self._cache(session_pca, func_memory_level=1))(
```

Comment:
You probably want to ignore a few parameters, like memory, memory_level and verbose in the _cache call.